planner/cascades/transformation_rules.go

```diff
@@ -323,3 +327,95 @@ func (r *PushSelDownProjection) OnTransform(old *memo.ExprIter) (newExprs []*mem
 	newTopSelExpr.SetChildren(newProjGroup)
 	return []*memo.GroupExpr{newTopSelExpr}, true, false, nil
 }
+
+// PushSelDownAggregation pushes Selection down to the child of Aggregation.
+type PushSelDownAggregation struct {
+}
+
+// GetPattern implements Transformation interface.
+// The pattern of this rule is `Selection -> Aggregation`.
+func (r *PushSelDownAggregation) GetPattern() *memo.Pattern {
+	return memo.BuildPattern(
+		memo.OperandSelection,
+		memo.EngineTiDBOnly,
+		memo.NewPattern(memo.OperandAggregation, memo.EngineTiDBOnly),
+	)
+}
+
+// Match implements Transformation interface.
+func (r *PushSelDownAggregation) Match(expr *memo.ExprIter) bool {
+	// The aggregation must be a complete mode.
+	agg := expr.Children[0].GetExpr().ExprNode.(*plannercore.LogicalAggregation)
+	for _, aggFunc := range agg.AggFuncs {
+		if aggFunc.Mode != aggregation.CompleteMode {
```

Comment:
Why do we require it?

Reply:
In https://github.com/pingcap/tidb/pull/13049, we may split the Aggregation and push the partial Aggregation down TableGather. If we don't require this, we may have this `finalAgg->Selection->TableGather->paritialAgg` when the `Selection` contains expressions that cannot be pushed down.

Reply:
And I don't know if this plan can be executed.

Reply:
I think this plan is correct since we can only push down selections that only contains groupby cols?

Reply:
Yes, the plan is correct. But I'm not sure if this plan can be correctly executed by the executor, since we have never generated such a plan before. I will add test for it after #13049 merged. For now, I'm ok to remove this check.